tests/lib/rules/require-meta-docs-recommended.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const rule = require('../../../lib/rules/require-meta-docs-recommended');
+const RuleTester = require('../eslint-rule-tester').RuleTester;
+
+const ruleTester = new RuleTester({
+  languageOptions: { sourceType: 'commonjs' },
+});
+
+ruleTester.run('require-meta-docs-recommended', rule, {
+  valid: [
+    'foo()',
+    'module.exports = {};',
+    `
+      module.exports = {
+        meta: { docs: { recommended: true } },
+        create(context) {}
+      };
+    `,
+    {
+      code: `
+        export default {
+          meta: { docs: { recommended: 'disallow unused variables' } },
```

Comment:
Should we be requiring this property to be a boolean?

Reply:
I would prefer not - in typescript-eslint we have it as a string or object.

Reply:
What are the use cases for that? I'm OK to leave as-is if you want, but personally, I've only ever seen it as a boolean.

Reply:
What would that look like on the typescript-eslint side? `meta.docs.config` with the `ConfigKey | Record<ConfigKey, any>` that have now?

Reply:
Ah, I just found/remembered another config that uses a non-boolean: `eslint-plugin-vitest`. https://github.com/veritem/eslint-plugin-vitest/blob/288bfb5b27fdfd838daef91b974a2484006db9c9/src/rules/expect-expect.ts#L41. Though, I don't see this reflected anywhere - maybe this isn't relevant...

Anyway, what I'm pondering now is: it feels to me like `meta.docs.recommended` operates as a subset of what e.g. `meta.docs.config` has to offer. As in, `docs: { recommended: true }` is roughly the same as `docs: { config: "recommended" }`, just ever so slightly fewer characters. But `config` allows plugins to have multiple "flavors". Which is a more extensible option. But is a rename...

Reply:
Yeah, I agree the `recommended` property can only represent a limited subset of potential information.

To represent rules that may belong to multiple configs or even have different severities in different configs, you probably need an object like:

```
configs: {
  recommend: true, // or perhaps a severity string like error/warn/off
  strict: ...
}
```

This is why I personally don't try to represent config information in the rules themselves except for plugins with a very simple config structure, and why I wanted to be sure we mentioned caveats to this rule like in https://github.com/eslint-community/eslint-plugin-eslint-plugin/pull/447#discussion_r1493366564

Reply:
Thanks, this conversation has been helpful. I just added an opt-in `allowNonBoolean` option.

Reply:
possible the option "allowedValues", it defaults to `[true, false]`;

The advantage of this is that when you configure ["strict"], if there is a spelling mistake(e.g. `"stricts"`) it will be caught.

Reply:
That wouldn't be useful to us in typescript-eslint, since we have config objects.

Reply:
`allowedValues` sounds like overkill to me. For most, it's a boolean, but if not, just let them specify whatever...